### PR TITLE
Enable `Client.RateLimits` to bypass the rate limit check

### PR DESCRIPTION
**Pull request by sa-spag**
Fixes https://github.com/google/go-github/issues/1899.

### CODE REVIEW

**Review by gmlewis — APPROVED** [review]
Very nice, @sa-spag !
I really like this solution... I was concerned that there would be much more disruption, but this looks very nice and clean... thank you!
LGTM.

Awaiting second LGTM before merging.

**Comment by gmlewis** [comment]
Whups, sorry, it looks like the unit tests are not passing:

```
    github_test.go:1708: rate.Reset.Time > now RateLimits resp = &http.Response{Status:"404 Not Found", StatusCode:404, Proto:"HTTP/1.1", ProtoMajor:1, ProtoMinor:1, Header:http.Header{"Content-Length":[]string{"19"}, "Content-Type":[]string{"text/plain; charset=utf-8"}, "Date":[]string{"Tue, 22 Jun 2021 17:21:16 GMT"}, "X-Content-Type-Options":[]string{"nosniff"}}, Body:io.nopCloser{Reader:(*bytes.Buffer)(0xc0003c1860)}, ContentLength:19, TransferEncoding:[]string(nil), Close:false, Uncompressed:false, Trailer:http.Header(nil), Request:(*http.Request)(0xc0002f5e00), TLS:(*tls.ConnectionState)(nil)}, want StatusCode=403
--- FAIL: TestRateLimits_coverage (0.00s)
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
Changing approval until unit tests pass.

**Comment by sa-spag** [comment]
> Very nice, @sa-spag !
> I really like this solution... I was concerned that there would be much more disruption, but this looks very nice and clean... thank you!

Credits to @ViBiOh for the `context.WithValue` trick I had no clue about.

> Whups, sorry, it looks like the unit tests are not passing...

Whups indeed, my bad. Got a little too confident, it is now fixed.

File: github/github_test.go
Comment (gmlewis) [inline github/github_test.go:205]: I'm concerned that we are not checking the value of `err` here... might there be a case where we get an error that will be  ignored here?

Comment (sa-spag) [reply github/github_test.go:205]: This helper explicitly sets the client up to trigger errors, and at this stage a rate limit error. As is, the test would fail if we check for `err != nil` for `methodName == "RateLimits"` as the request would result in a HTTP 404, since no handler has been created. We can set one up on `TestRateLimits_coverage` but I think it makes the test less straightforward, and the nominal case is already covered by other functions.

**Review by gmlewis — APPROVED** [review]
LGTM.

Awaiting second LGTM before merging.

**Comment by gmlewis** [comment]
Friendly ping @wesleimp

**Review by Parker77 — APPROVED** [review]
LGTM.

**Pull request merged**
